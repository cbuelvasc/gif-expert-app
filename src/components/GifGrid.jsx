import PropTypes from "prop-types";
import { GifItem, LoadingGifs } from "./";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h2 className="text-2xl font-bold tracking-tight text-gray-900 mt-6">
        {category}
      </h2>
      {isLoading && <LoadingGifs />}
      <div className="mt-6 grid grid-cols-1 gap-x-6 gap-y-10 sm:grid-cols-2 lg:grid-cols-4 xl:gap-x-8">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
