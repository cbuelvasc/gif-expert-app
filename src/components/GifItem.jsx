import PropTypes from "prop-types";

export const GifItem = ({ title, url }) => {
  return (
    <div className="group relative shadow-lg">
      <div className="aspect-h-1 aspect-w-1 w-full overflow-hidden rounded-md bg-gray-200 lg:aspect-none group-hover:opacity-75 lg:h-80 ">
        <img
          src={url}
          alt={title}
          className="h-full w-full object-cover object-center lg:h-full lg:w-full"
        />
      </div>
      <div className="mt-4 flex justify-between">
        <div>
          <h3 className="text-sm text-gray-700 pl-2 ">
            <a href="#">
              <span aria-hidden="true" className="absolute inset-0" />
              {title}
            </a>
          </h3>
        </div>
      </div>
      <div className="mt-4 flex justify-between"></div>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
