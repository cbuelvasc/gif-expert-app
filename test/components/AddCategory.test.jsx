/* eslint-disable no-undef */
import { fireEvent, render, screen } from "@testing-library/react";
import { AddCategory } from "../../src/components/AddCategory";

describe("Testing for <AddCategory />", () => {
  const onNewCategory = jest.fn();
  const inputText = "One Punch";

  test("should be matched with the snapshot", () => {
    const { container } = render(<AddCategory onNewCategory={onNewCategory} />);

    expect(container).toMatchSnapshot();
  });

  test("should be changed the input value", () => {
    render(<AddCategory onNewCategory={onNewCategory} />);
    const input = screen.getByRole("textbox");

    fireEvent.change(input, { target: { value: inputText } });

    expect(input.value).toBe(inputText);
  });

  test("should display the input with the placeholder", () => {
    render(<AddCategory onNewCategory={onNewCategory} />);
    const input = screen.getByRole("textbox");

    expect(input).toBeTruthy();
    expect(input.placeholder).toBe("Type anything...");
  });

  test("should call the onNewCategory function when submitting the form", () => {
    render(<AddCategory onNewCategory={onNewCategory} />);
    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.change(input, { target: { value: inputText } });
    fireEvent.submit(form);

    expect(input.value).toBe("");
    expect(onNewCategory).toHaveBeenCalled();
    expect(onNewCategory).toHaveBeenCalledTimes(1);
    expect(onNewCategory).toHaveBeenCalledWith(inputText);
  });

  test("should not call the onNewCategory function when submitting the form with an empty input", () => {
    const onNewCategory = jest.fn();
    render(<AddCategory onNewCategory={onNewCategory} />);
    const form = screen.getByRole("form");

    fireEvent.submit(form);

    expect(onNewCategory).toHaveBeenCalledTimes(0);
    expect(onNewCategory).not.toHaveBeenCalled();
  });
});
