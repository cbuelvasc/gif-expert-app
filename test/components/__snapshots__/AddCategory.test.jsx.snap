// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing for <AddCategory /> should be matched with the snapshot 1`] = `
<div>
  <form
    aria-label="form"
    class="group relative"
  >
    <svg
      aria-hidden="true"
      class="absolute left-3 top-1/2 -mt-2.5 text-slate-400 pointer-events-none group-focus-within:text-blue-500"
      fill="currentColor"
      height="20"
      width="20"
    >
      <path
        clip-rule="evenodd"
        d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
        fill-rule="evenodd"
      />
    </svg>
    <input
      aria-label="Type anything..."
      class="focus:ring-2 focus:ring-blue-500 focus:outline-none appearance-none w-full text-sm leading-6 text-slate-900 placeholder-slate-400 rounded-md py-2 pl-10 ring-1 ring-slate-200 shadow-xl"
      placeholder="Type anything..."
      type="text"
      value=""
    />
  </form>
</div>
`;
